sqlglot/optimizer/merge_subqueries.py

```diff
@@ -167,6 +167,7 @@ def _outer_select_joins_on_inner_select_join():
         and isinstance(inner_select, exp.Select)
         and not any(inner_select.args.get(arg) for arg in UNMERGABLE_ARGS)
         and inner_select.args.get("from")
+        and not outer_scope.pivots
```

Comment:
We cant merge subqueries that are pivoted, because again we'll lose the `pivot` arg. For example:

```python
>>> from sqlglot.optimizer import optimize
>>> print(optimize("SELECT * FROM (SELECT a, b, c FROM sc.tb) PIVOT (SUM(c) FOR b IN ('x','y','z'))").sql(pretty=True))
SELECT
  "_q_1"."a" AS "a",
  "_q_1"."x" AS "x",
  "_q_1"."y" AS "y",
  "_q_1"."z" AS "z"
FROM "sc"."tb" AS "tb"  -- subquery was merged and we lost the pivot
```